modules/dkimPolicy.jsm

```diff
@@ -525,6 +525,7 @@ var Policy = {
 			}
 
 			// Check if enabled for SDID.
+			if (!sdid) { return undefined; }
```

Comment:
Not that on master this check is done in `addFavicons()` in `AuthVerifier.jsm`.

I'm fine with leaving the check here, but then you should adapt the type of the `sdid` parameter to `String|undefined`.

Reply:
Thank you for your guidance, again. I moved the check to AuthVerifier.jsm, in the end.
I guess, adding/updating type checks to the 2.x branch would be a lot of work, as I didn't really care about them. So thank you for your kind offer. But I'm not sure, I want to do this at the moment.

Off topic, but no problem:
- I like my configuration of Thunderbird with all its addons and the theme, as it is. At the moment I'm willing to put some effort in improving it. I've fixed numerous issues, which came to my attention. But mostly for addons, where the developer lost his/her interest in keeping up with the changes in Thunderbird. 
- I learn a lot, while I'm doing this (which would be a point on the pro site of adding type checking to the 2.x branch)
- I know, there are more and more security issues on the other hand. This is the only reason I can think of, at the moment, which could bring me to switch to a current mail client.
- But changing the mail client wouldn't be easy; probably I wouldn't find a client with all the features, I'm used to. There aren't many mail clients on windows, so the options are pretty limited. The next redesign for the current Thunderbird is already announced - but probably the developers will announce another redesign before I can invest some time to try to get it in a usable state (usable in my eyes)